### PR TITLE
feat: new image parsing on Environment page

### PR DESCRIPTION
**Pull request by agatha197**
related PR: https://github.com/lablup/backend.ai/pull/2939

# Extended Image Information Support

This PR adds support for extended image information in the ImageList component and related translations. It introduces new fields and modifies the display of image details when the backend supports the 'extended-image-info' feature.

**Checklist:**
- [ ] Minimum required manager version: 24.09.1

## Changes

1. Updated `ImageListQuery` to include new fields: `namespace`, `base_image_name`, `tags`, and `version`.
2. Modified the ImageList component to conditionally render columns based on the `supportExtendedImageInfo` flag.
3. Added new translations for "Base image name" and "Tags" in multiple languages.
4. Updated the backend client to support the 'extended-image-info' feature for manager version 24.09.1 and above.

## Impact

- Users will see more detailed image information when using a compatible backend version.
- The image list will display new columns for base image name and tags when supported.
- Existing functionality is preserved for older backend versions.

## Review Notes

- Verify that the image list displays correctly with both old and new backend versions.
- Check that new translations are applied correctly in different languages.
- Ensure that sorting and filtering work properly with the new fields.

## How to test
1. Checkout core branch to `topic/10-22-feature_add_info_field_to_gql_image_schema` https://github.com/lablup/backend.ai/pull/2993
2. Go to Environment page. 

## What to check:
- [ ] Check the data is valid.
  - [ ]  Since 24.09.1: Full image path, Registry, Architecture, Namespace, Base image name, Version, Tags, Digest, Resource limit, Control.
  - [ ] Before 24.09.1: Full image path, Registry, Architecture, Name, Language, Version, Base, Constraints, Digest, Resource limit, Control.
- [ ] All data is highlightable by searching images.
- [ ] Sortable data works fine.

### CODE REVIEW

**Comment by agatha197** [comment]
* **#2785** <a href="https://app.graphite.dev/github/pr/lablup/backend.ai-webui/2785?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @agatha197 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: react/src/components/ImageList.tsx [hunk 1]
```diff
@@ -118,140 +139,182 @@ const ImageList: React.FC<{ style?: React.CSSProperties }> = ({ style }) => {
...
+                b?.tag && b?.labels
+                  ? getConstraints(
+                      b?.tag,
+                      b?.labels as { key: string; value: string }[],
+                    )[0] || ''
+                  : '';
+              if (requirementA === '' && requirementB === '') return 0;
+              if (requirementA === '') return -1;
+              if (requirementB === '') return 1;
+              return requirementA.localeCompare(requirementB);
```
Comment (ironAiken2) [inline react/src/components/ImageList.tsx]:
```suggestion
return localeCompare(requirementA, requirementB);
```

File: react/src/components/ImageList.tsx [hunk 2]
```diff
@@ -118,140 +139,182 @@ const ImageList: React.FC<{ style?: React.CSSProperties }> = ({ style }) => {
...
+            ),
+          },
+          {
+            title: t('environment.Language'),
+            key: 'lang',
+            dataIndex: 'lang',
+            sorter: (a: EnvironmentImage, b: EnvironmentImage) => {
+              const langA = a?.name ? getLang(a?.name) : '';
+              const langB = b?.name ? getLang(b?.name) : '';
+              return langA && langB ? langA.localeCompare(langB) : 0;
```
Comment (ironAiken2) [inline react/src/components/ImageList.tsx]:
```suggestion
            return localeCompare(
                getLang(a.name ?? ''),
                getLang(b.name ?? ''),
              );
```

File: react/src/components/ImageList.tsx [hunk 3]
```diff
@@ -118,140 +139,182 @@ const ImageList: React.FC<{ style?: React.CSSProperties }> = ({ style }) => {
...
+          {
+            title: t('environment.Namespace'),
+            key: 'name',
+            dataIndex: 'name',
+            sorter: (a: EnvironmentImage, b: EnvironmentImage) => {
+              const namespaceA = getNamespace(getImageFullName(a) || '');
+              const namespaceB = getNamespace(getImageFullName(b) || '');
+              return namespaceA && namespaceB
+                ? namespaceA.localeCompare(namespaceB)
+                : 0;
```
Comment (ironAiken2) [inline react/src/components/ImageList.tsx]:
```suggestion
return localeCompare(getImageFullName(a), getImageFullName(b));
```

**Review by ironAiken2 — CHANGES_REQUESTED** [review]
It works fine but I left few small comments :)

File: react/src/components/ImageList.tsx [hunk 4]
```diff
@@ -52,13 +63,16 @@ const ImageList: React.FC<{ style?: React.CSSProperties }> = ({ style }) => {
   const [imageSearch, setImageSearch] = useState('');
   const [isPendingRefreshTransition, startRefreshTransition] = useTransition();
   const [isPendingSearchTransition, startSearchTransition] = useTransition();
+  const baiClient = useSuspendedBackendaiClient();
+  const supportExtendedImageInfo =
+    baiClient?.supports('extended-image-info') ?? false;
```
Comment (ironAiken2) [inline react/src/components/ImageList.tsx]:
```suggestion
  const supportExtendedImageInfo =
    baiClient?.supports('extended-image-info');
```

**Review by ironAiken2 — APPROVED** [review]
(dismissed)

**Review by ironAiken2 — CHANGES_REQUESTED** [review]
Table Column name 을 대문자로 시작하도록 바꿔주세요.. 놓쳐서 죄송합니다. 

e.g.) `Full image path` -> `Full Image Path`

File: react/src/hooks/index.tsx [hunk 5]
```diff
@@ -310,18 +310,17 @@ export const useBackendAIImageMetaData = () => {
...
-        return metadataTagAlias || tag;
+            return pattern.test(tag)
+              ? _.startCase(_.replace(tag, pattern, replaceString))
+              : acc;
+          },
+          _.startCase(tag),
+        );
+        return metadataTagAlias;
       },
     },
```
Comment (yomybaby) [inline react/src/hooks/index.tsx]: How about this suggestion for readability and less excuting Regexp test? 
```suggestion
      tagAlias: (tag: string) => {
        return (
          metadata?.tagAlias[tag] ??
          _.chain(metadata.tagReplace)
            .toPairs()
            .find(([regExpStr]) => new RegExp(regExpStr).test(tag))
            .thru((pair) => {
              if (pair) {
                const [regExpStr, replaceStr] = pair;
                return _.replace(tag, new RegExp(regExpStr), replaceStr);
              }
            })
            .value() ??
          _.startCase(tag)
        );
      },
    },
```

**Review by yomybaby — CHANGES_REQUESTED** [review]
Please check my suggestion.

**Review by yomybaby — APPROVED** [review]
LGTM

**Comment by agatha197** [comment]
> Please change the Table Column name to start with an uppercase letter... sorry I missed it.
> 
> e.g.) `Full image path` -> `Full Image Path`

I followed Backend.AI Writing Rules 😉

**Review by ironAiken2 — APPROVED** [review]
LGTM

**Pull request merged**